legacy/src/app/directives/tests/test_machines_table.js

```diff
@@ -1117,4 +1115,42 @@ barbaz`); // Has to be formatted this way for tooltip
       expect(scope.getLimit(group)).toEqual(undefined);
     });
   });
+
+  describe("nodeSelectionNotActionable", () => {
+    it("returns true if at least one node in selection cannot perform action", () => {
+      const directive = compileDirective();
+      const scope = directive.isolateScope();
+      const machines = Array.from(Array(3)).map(makeMachine);
+      machines[0].actions = ["action1", "action2"];
+      machines[1].actions = ["action1", "action2"];
+      machines[2].actions = ["action1"];
+      scope.actionOption = { name: "action2" };
+
+      expect(scope.nodeSelectionNotActionable(machines)).toEqual(true);
+    });
+
+    it("returns false if all nodes in selection can perform action", () => {
+      const directive = compileDirective();
+      const scope = directive.isolateScope();
+      const machines = Array.from(Array(3)).map(makeMachine);
```

Comment:
Am I being dumb? Why make a new copy of an empty array that you already have?
```suggestion
      const machines = Array(3).map(makeMachine);
```

Reply:
You're not being dumb, I'm just being copy+pastey. `Array(3).map(makeMachine)` won't work though because the array is actually empty - a spooky ghost array that has a length of 3 *and nothing else*, so it can't actually be mapped. I think that's what the original code was trying to get around, but it's probably easier and more readable for it to be `Array(3).fill().map(makeMachine)` which fills it with something iterable first.